MPI changes (#47)
nesi/perf-training: _lessons/python-scatter/05-mpi.md

 * there are no serial sections in MPI code and hence MPI programs need to be written to run in parallel from the beginning to the end
 * some algorithms are difficult to implement efficiently with MPI due its distributed memory approach and communication overheads
 * it is easy to create "dead locks" where one or more processes get stuck waiting for a message

>  * specified computing resources are occupied the whole run time, no dynamic allocation/deallocation of CPU cores

Reviewer: ```suggestion
 * the requested computing resources are occupied the whole run time, i.e. there is no dynamic allocation/deallocation of CPU cores
```

## How to use MPI to accelerate `scatter.py`

 * `from mpi4py import MPI` at the top. This will initialise MPI.

>  * `comm = MPI_COMM_WORLD` sets the communicator.

Reviewer: ```suggestion
 * `comm = MPI.COMM_WORLD` gets the communicator.
```

>  * `nprocs = comm.Get_size()` stores the total number of processes.

Reviewer: ```suggestion
 * `nprocs = comm.Get_size()` gets the total number of processes.
```

>  * `pe = comm.Get_rank()` stores the actual rank of an process.

Reviewer: ```suggestion
 * `pe = comm.Get_rank()` gets the actual rank of a process.
```

Author: these lines get the comm/rank/... and store it an variable. So one could argue, but I am fine with the change.
